gobblin-core/src/main/java/org/apache/gobblin/converter/parquet/JsonElementConversionFactory.java

```diff
@@ -0,0 +1,476 @@
+     * Set a {@link JsonElementConverter} for the elements within a collection.
+     * @param elementConverter
+     */
+    protected void setElementConverter(JsonElementConverter elementConverter) {
+      this.elementConverter = elementConverter;
+    }
+
+    public JsonElementConverter getElementConverter() {
+      return this.elementConverter;
+    }
+  }
+
+  public static class IntConverter extends PrimitiveConverter {
+
+    public IntConverter(JsonSchema schema, boolean repeated) {
+      super(schema, repeated, INT32);
+    }
+
+    @Override
+    IntegerValue convertField(JsonElement value) {
+      return new IntegerValue(value.getAsInt());
+    }
+  }
+
+  public static class LongConverter extends PrimitiveConverter {
+
+    public LongConverter(JsonSchema schema, boolean repeated) {
+      super(schema, repeated, INT64);
+    }
+
+    @Override
+    LongValue convertField(JsonElement value) {
+      return new LongValue(value.getAsLong());
+    }
+  }
+
+  public static class FloatConverter extends PrimitiveConverter {
+
+    public FloatConverter(JsonSchema schema, boolean repeated) {
+      super(schema, repeated, PrimitiveTypeName.FLOAT);
+    }
+
+    @Override
+    FloatValue convertField(JsonElement value) {
+      return new FloatValue(value.getAsFloat());
+    }
+  }
+
+  public static class DoubleConverter extends PrimitiveConverter {
+
+    public DoubleConverter(JsonSchema schema, boolean repeated) {
+      super(schema, repeated, PrimitiveTypeName.DOUBLE);
+    }
+
+    @Override
+    DoubleValue convertField(JsonElement value) {
+      return new DoubleValue(value.getAsDouble());
+    }
+  }
+
+  public static class BooleanConverter extends PrimitiveConverter {
+
+    public BooleanConverter(JsonSchema schema, boolean repeated) {
+      super(schema, repeated, PrimitiveTypeName.BOOLEAN);
+    }
+
+    @Override
+    BooleanValue convertField(JsonElement value) {
+      return new BooleanValue(value.getAsBoolean());
+    }
+  }
+
+  public static class StringConverter extends PrimitiveConverter {
+
+    public StringConverter(JsonSchema schema, boolean repeated) {
+      super(schema, repeated, BINARY);
+    }
+
+    @Override
+    BinaryValue convertField(JsonElement value) {
+      return new BinaryValue(Binary.fromString(value.getAsString()));
+    }
+
+    @Override
+    public Type schema() {
+      String columnName = this.jsonSchema.getColumnName();
+      if (this.repeated) {
+        return Types.repeated(BINARY).as(UTF8).named(columnName);
+      }
+      switch (this.jsonSchema.optionalOrRequired()) {
+        case OPTIONAL:
+          return Types.optional(BINARY).as(UTF8).named(columnName);
+        case REQUIRED:
+          return Types.required(BINARY).as(UTF8).named(columnName);
+        default:
+          throw new RuntimeException("Unsupported Repetition type");
+      }
+    }
+  }
+
+  public static class ArrayConverter extends CollectionConverter {
+    private final JsonSchema elementSchema;
```

Comment:
I don't see the necessity to make an extra `elementSchema` field, as the `elementConverter` has inner reference to it. It's applied to `EnumConverter`.